meson.build

```diff
@@ -371,6 +371,16 @@ if librsvg_found
     cfg_var.set('HAVE_RSVG', '1')
 endif
 
+# libresvg doesn't ship pkg-config files, so we can't use dependency().
+libresvg_dep = cc.find_library('resvg', has_headers: ['resvg.h'], required: get_option('resvg'))
+libresvg_found = libresvg_dep.found()
+if libresvg_found
+    # It's annoying that I have to spell out the `include_directories` here, but without it Meson can build
+    # but `compile_commands.json` doesn't have the -I$PREFIX/include, making code analysis tools & VSCode fail.
+    libvips_deps += declare_dependency(dependencies: [libresvg_dep], include_directories: [include_directories(get_option('prefix') / get_option('includedir'))], link_args: ['-ldl'])
```

Comment:
Hmm, weird. I assume this is on Windows or macOS (due to the use of `-ldl`)? Perhaps compiling with `CFLAGS=" -I$PREFIX/include"` would be an alternative way to fix this?

Reply:
Alternatively, we could add `dependency()` construction and generate the `resvg.pc` manually, similar to what we do with PDFium, see: https://github.com/libvips/libvips#pdfium.

Reply:
> and generate the resvg.pc manually

Is it worth it / is the current implementation that bad?

Reply:
And no, I haven't tried to build it on Windows / macOS, it was either on Linux or Emscripten that I needed that.

Reply:
> Is it worth it / is the current implementation that bad?

I'm not sure, I think this is also fine. I'm only a bit worried that I broke VSCode and code analysis tools with commit https://github.com/kleisauke/libvips/commit/349a7b766333f831790374fe06ca21d64d8b26ee, which avoids this `declare_dependency()` construction altogether and just uses `cc.find_library()` instead.

Reply:
Might be worth checking compile_commands.json for presence of this `-I`. I think the problem was that otherwise Meson would pass include directory via environment variable, which obviously wouldn't be reflected in compile_commands.json explicit flags.

OTOH if you don't use autocomplete or code analysis tools, maybe it doesn't matter much. For me it was particularly useful for debugging while working on the integration though.

Reply:
I see, I'll investigate.

Reply:
I could fix the compilation database (`compile_commands.json`) with commit cbc753c70289ed67e641c779170754c3c5d5dbd4 (merged with commit 17091e6ce19c263f885997f1434551e281099d2c).

Tested with (in combination with CLion):
<details>
  <summary>Details</summary>

Build and install resvg with:
```console
$ git clone https://github.com/RazrFalcon/resvg
$ cd resvg/c-api/
$ cargo build --release
$ sudo cp ../target/release/libresvg.so /usr/lib64/
$ sudo cp resvg.h /usr/include/
```

Build and install libvips with:
```console
$ CC="gcc" CXX="g++" meson setup build --prefix=/usr --buildtype=release -Ddeprecated=false -Drsvg=disabled -Dmodules=disabled
$ meson compile -Cbuild
$ sudo meson install -Cbuild
```

Note: `CC="gcc" CXX="g++"` is [due to broken compiler detection in CLion for ccache](https://youtrack.jetbrains.com/issue/CPP-11255/Using-ccache-prevents-compiler-detection-in-CLion).
</details>

Though, that installs the libraries in `/usr`, which may not be desirable. 

Note that I still don't see the `-I/usr/include` in the compilation database, so I suspect that this issue may also occur when installing the libraries in a non-standard prefix.

<details>
  <summary>Details</summary>

```console
$ grep -o '\-I/usr/include/\S*' build/compile_commands.json | sort --unique
-I/usr/include/blkid
-I/usr/include/cairo
-I/usr/include/cfitsio
-I/usr/include/freetype2
-I/usr/include/fribidi
-I/usr/include/glib-2.0
-I/usr/include/harfbuzz
-I/usr/include/ImageMagick-6
-I/usr/include/Imath
-I/usr/include/libgsf-1
-I/usr/include/libmount
-I/usr/include/libpng16
-I/usr/include/libvmaf
-I/usr/include/libxml2
-I/usr/include/nifti
-I/usr/include/OpenEXR
-I/usr/include/openjpeg-2.5
-I/usr/include/openslide
-I/usr/include/pango-1.0
-I/usr/include/pixman-1
-I/usr/include/poppler
-I/usr/include/poppler/glib
-I/usr/include/rav1e
-I/usr/include/svt-av1
-I/usr/include/sysprof-4
-I/usr/include/webp
```
</details>